libs/ledger-live-common/src/families/cosmos/js-buildTransaction.ts

```diff
@@ -43,193 +65,239 @@ export const buildTransaction = async (
+        aminoMsgs.push(aminoMsg);
+
+        // PROTO MESSAGE
+        protoMsgs.push({
+          typeUrl: "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
+          value: MsgWithdrawDelegatorReward.encode({
+            delegatorAddress: account.freshAddress,
+            validatorAddress: validator.address,
+          }).finish(),
         });
       }
       break;
-
     case "claimRewardCompound":
       if (
-        !transaction.validators ||
-        transaction.validators.length < 1 ||
-        !transaction.validators[0].address ||
-        transaction.validators[0].amount.lte(0)
+        transaction.validators &&
+        transaction.validators.length > 0 &&
+        transaction.validators[0].address &&
+        transaction.validators[0].amount.gt(0)
       ) {
-        isComplete = false;
-      } else {
-        msg.push({
-          typeUrl: "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
+        const validator = transaction.validators[0];
+        // AMINO MESSAGES
+        const aminoWithdrawRewardMsg: AminoMsgWithdrawDelegatorReward = {
+          type: "cosmos-sdk/MsgWithdrawDelegationReward",
           value: {
-            delegatorAddress: account.freshAddress,
-            validatorAddress: transaction.validators[0].address,
+            delegator_address: account.freshAddress,
+            validator_address: validator.address,
           },
-        });
+        };
+        const aminoDelegateMsg: AminoMsgDelegate = {
+          type: "cosmos-sdk/MsgDelegate",
+          value: {
+            delegator_address: account.freshAddress,
+            validator_address: validator.address,
+            amount: {
+              denom: account.currency.units[1].code,
+              amount: validator.amount.toString(),
+            },
+          },
+        };
+        aminoMsgs.push(aminoWithdrawRewardMsg, aminoDelegateMsg);
 
-        msg.push({
+        // PROTO MESSAGES
+        protoMsgs.push({
+          typeUrl: "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
+          value: MsgWithdrawDelegatorReward.encode({
+            delegatorAddress: account.freshAddress,
+            validatorAddress: validator.address,
+          }).finish(),
+        });
+        protoMsgs.push({
           typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
-          value: {
+          value: MsgDelegate.encode({
             delegatorAddress: account.freshAddress,
-            validatorAddress: transaction.validators[0].address,
+            validatorAddress: validator.address,
             amount: {
               denom: account.currency.units[1].code,
-              amount: transaction.validators[0].amount.toString(),
+              amount: validator.amount.toString(),
             },
-          },
+          }).finish(),
         });
       }
       break;
   }
-
-  if (!isComplete) {
-    return [];
-  }
-
-  return msg;
+  return { aminoMsgs, protoMsgs };
 };
 
 export const postBuildTransaction = async (
```

Comment:
Can you please also comment here to explain what this function does / why we need it?